src/connector/src/parser/mod.rs

```diff
@@ -423,6 +430,17 @@ impl SpecificParserConfig {
             SourceFormat::Maxwell => SpecificParserConfig::Maxwell,
             SourceFormat::CanalJson => SpecificParserConfig::CanalJson,
             SourceFormat::Native => SpecificParserConfig::Native,
+            SourceFormat::DebeziumAvro => {
+                debug_assert!(info.use_schema_registry);
```

Comment:
avro can be used without schema registry

Reply:
debezium can be used without schema registry if JSON is used as the data format, which is the default option in debezium. Also, debezium remove its avro dependencies starting from version 2.0.0. However, mz doesn't even support JSON.

For AVRO data format, techinically it is possible to be used in debezium without schema registry if one provides the schema file and scarifies schema evolution but I doubt whether there is any real-world use cases. Therefore, I suggest we enforce schema registry for debezium avro and add back the schema file support only when needed in order no to confuse user.